airflow/operators/subdag.py

```diff
@@ -112,7 +113,9 @@ def _validate_pool(self, session):
             conflicts = [t for t in self.subdag.tasks if t.pool == self.pool]
             if conflicts:
                 # only query for pool conflicts if one may exist
-                pool = session.query(Pool).filter(Pool.slots == 1).filter(Pool.pool == self.pool).first()
+                pool = session.scalar(
+                    select(Pool).where(Pool.slots == 1).where(Pool.pool == self.pool).limit(1)
+                )
```

Comment:
```suggestion
                pool = session.scalar(select(Pool).where(Pool.slots == 1).where(Pool.pool == self.pool))
```

Since `pool` is unique